optimum/neuron/accelerate/state.py

```diff
@@ -84,6 +90,11 @@ def __init__(self, cpu: bool = False, **kwargs):
                         self.device = torch.device("cuda", self.local_process_index)
                     torch.cuda.set_device(self.device)
             elif is_torch_xla_available() and not cpu:
+                # It is important to set the environment variables before initializing the process group otherwise they will be ignored by the Neuron compiler.
+                set_common_neuron_cc_flags()
+                if os.environ.get("ACCELERATE_USE_AMP", "false") == "true":
+                    set_neuron_cc_flags_for_torch_amp()
```

Comment:
I looked at the AWS documentation and they don't seem to care much either about restoring env variable.
Forget about my comment.

Reply:
In any case, as explained, it does not really change anything, once the process group has been initialized, we cannot change the environment for the Neuron compiler. That is also the reason why I changed some of the ways we set the flags. Any flag that is model dependent cannot be set by `optimum-neuron` because by the time we have the model, we usually already have initialized the process group.